connectors/src/api/webhooks/webhook_slack.ts

```diff
@@ -353,14 +361,47 @@ const _webhookSlackAPIHandler = async (
                 });
               }
               try {
-                await Promise.all(
-                  activeConfigurations.map((c) =>
-                    updateSlackChannelInConnectorsDb({
+                await concurrentExecutor(
+                  activeConfigurations,
+                  async (c) => {
+                    const connector = await ConnectorResource.fetchById(
+                      c.connectorId
+                    );
+                    if (!connector) {
+                      return apiError(req, res, {
+                        api_error: {
+                          type: "connector_not_found",
+                          message: `Connector ${c.connectorId} not found`,
+                        },
+                        status_code: 404,
+                      });
```

Comment:
let's not send an apiError inside a parallel execution loop - we may send send errors multiple times, which will crash, and we won't see the error anyway. A log should be enough